GooglePlayInstant/Tests/Editor/QuickDeploy/ConfigurationTest.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.IO;
+using GooglePlayInstant.Editor.QuickDeploy;
+using GooglePlayInstant.LoadingScreen;
+using NUnit.Framework;
+using UnityEditor;
+using UnityEngine;
+
+namespace GooglePlayInstant.Tests.Editor.QuickDeploy
+{
+    public class ConfigurationTest
+    {
+        private static readonly string TestConfigurationPath =
+            Path.Combine("Assets", LoadingScreenConfig.EngineConfigurationFileName);
+
+        // Dispose of temporarily created file.  
+        [TearDown]
+        public void Cleanup()
+        {
+            AssetDatabase.DeleteAsset(TestConfigurationPath);
+        }
+
+        [Test]
+        public void TestSavingConfigOnCreateBundle()
+        {
+            var inputConfig = new QuickDeployConfig.EditorConfiguration();
+            QuickDeployConfig.SaveEditorConfiguration(QuickDeployWindow.ToolBarSelectedButton.CreateBundle, inputConfig,
+                TestConfigurationPath);
+
+            var outputConfigurationJson = File.ReadAllText(TestConfigurationPath);
+            var outputConfig =
+                JsonUtility.FromJson<QuickDeployConfig.EditorConfiguration>(outputConfigurationJson);
+
+            Assert.AreEqual(outputConfig.assetBundleFileName, QuickDeployConfig.AssetBundleFileName);
```

Comment:
For this and the other ones, since I don't see any actual strings, it's not clear to me what your checking. Is this checking that "" == "" ?

Consider defining some const strings to set and compare to. Both here and on the asserts below.

Reply:
Unit tests shouldn't depend on values that happen to be saved in the local environment.

Ideally you'd provide a temp location to save the config file as well as the values to put in the file and to assert when read.

Another thing that unit tests shouldn't do is modify the real environment (i.e. the real config file).

Reply:
Refactored to make Quick Deploy an instance to test these strings in response to this ^